src/NuGet.CommandLine/Commands/RestoreCommand.cs

```diff
@@ -159,108 +161,105 @@ public RestoreCommand()
             {
                 // Restore a project.json file using the directory as the Id
                 Console.LogVerbose($"Reading project file {Arguments[0]}");
-                var projectDirectory = Path.GetDirectoryName(inputPath);
 
                 projectJsonPath = inputPath;
-
-                packageSpec = JsonPackageSpecReader.GetPackageSpec(
-                    File.ReadAllText(projectJsonPath),
-                    Path.GetFileName(projectDirectory),
-                    projectJsonPath);
+                projectName = Path.GetFileName(projectDirectory);
             }
-            else if (MsBuildUtility.IsMsBuildBasedProject(inputPath))
+            else if (ProjectHelper.UnsupportedProjectExtensions.Contains(Path.GetExtension(inputPath)))
```

Comment:
That wouldn't be passed in as an input here. The exception to this is nuget.exe restore <project.json> where we do not read the project file at all, but that is caused by the user.